core/src/main/scala/zio/config/ConfigDocs.scala

```diff
@@ -7,10 +7,13 @@ object ConfigDocs {
 
   final case class Sources(set: Set[String])
 
-  final case object Empty                                                               extends ConfigDocs[Nothing, Nothing]
-  final case class NestedPath[K, V](path: K, docs: ConfigDocs[K, V])                    extends ConfigDocs[K, V]
-  final case class Both[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V])          extends ConfigDocs[K, V]
-  final case class OneOf[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V])         extends ConfigDocs[K, V]
-  final case class Sequence[K, V](element: List[ConfigDocs[K, V]])                      extends ConfigDocs[K, V]
-  final case class DynamicMap[K, V](element: scala.collection.Map[K, ConfigDocs[K, V]]) extends ConfigDocs[K, V]
+  final case object Empty                                                       extends ConfigDocs[Nothing, Nothing]
+  final case class NestedPath[K, V](path: K, docs: ConfigDocs[K, V])            extends ConfigDocs[K, V]
+  final case class Both[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V])  extends ConfigDocs[K, V]
+  final case class OneOf[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V]) extends ConfigDocs[K, V]
+  final case class Sequence[K, V](element: List[ConfigDocs[K, V]])              extends ConfigDocs[K, V]
+  final case class DynamicMapInit[K, V](element: ConfigDocs[K, V])              extends ConfigDocs[K, V]
+  // During reporting, we know the dynamic keys; Fixme; Make it better orthogonal
+  final case class DynamicMap[K, V](element: Map[K, ConfigDocs[K, V]]) extends ConfigDocs[K, V]
```

Comment:
Maybe:

```scala
DynamicMap[K, V](keys: List[K], values: ConfigDocs[K, V])
```

Then in reporting you can fill in `keys` but otherwise leave it empty.